kiosk/src/Components/AddingGame.tsx

```diff
@@ -48,15 +48,14 @@ const AddingGame: React.FC<IProps> = ({ kiosk }) => {
                 clearInterval(intervalId);
             }
         };
-
     });
 
     useEffect(() => {
         async function generateKioskCode() {
             try {
                 const newKioskCode: string = await generateKioskCodeAsync();
                 setKioskCode(newKioskCode);
-                await addGameToKiosk(newKioskCode);
+                await (await addGameToKiosk(newKioskCode));
```

Comment:
Why do you need a second await?

Reply:
I'm not sure how necessary it is, but it's because the `addGameToKiosk` function is now returning a promise that has an internal async function. I might not have the right understanding, though.

Reply:
promises collapse so only need one await (that is, you can think of a promise that returns a promise as if it implicitly awaited the returned promise)